### PR TITLE
Add support for Market Details page to Layout header

### PR DESCRIPTION
**Pull request by therealemjy**
This PR adds support for displaying a back button inside the `Layout` header when navigating to the Market Details page.

### CODE REVIEW

File: src/components/v2/Layout/Header/index.tsx [hunk 1]
```diff
@@ -19,19 +24,35 @@ const Header = () => {
...
+        </Link>
+      );
+    }
+
     const currentItem = menuItems.find(item => item.href === pathname);
     const currentItemKey = currentItem?.i18nTitleKey || currentItem?.i18nKey;
+
     if (!currentItemKey) {
-      console.error(`missed key for translation: ${pathname}`);
+      console.error(`Missing translation for key: ${pathname}`);
```
Comment (coreyar) [inline src/components/v2/Layout/Header/index.tsx]: We could leave a comment here to add error reporting which would help us in the future when we add better error reporting

File: src/components/v2/Layout/Header/index.tsx [hunk 2]
```diff
@@ -19,19 +24,35 @@ const Header = () => {
   const isOnMobile = useIsMdDown();
 
   const title = useMemo(() => {
+    // Handle special case of Market Details page
```
Comment (coreyar) [inline src/components/v2/Layout/Header/index.tsx]: If we write this as a component not in the render scope we wouldn't need to memoize it.

File: src/components/v2/Layout/Header/index.tsx (hunk 1 above)
Comment (therealemjy) [reply src/components/v2/Layout/Header/index.tsx]: Good idea.

File: src/components/v2/Layout/Header/index.tsx (hunk 2 above)
Comment (therealemjy) [reply src/components/v2/Layout/Header/index.tsx]: Yeah.
There's a few improvements that can be done here. I'll update

**Review by coreyar — APPROVED** [review]

**Pull request merged**